test/file.ts

```diff
@@ -2678,6 +2685,82 @@ describe('File', () => {
           });
         });
       });
+
+      it('should recursively create directory and write file contents if destination path is nested', done => {
+        tmp.setGracefulCleanup();
+        tmp.dir(async (err, tmpDirPath) => {
+          assert.ifError(err);
+
+          const fileContents = 'nested-abcdefghijklmnopqrstuvwxyz';
+
+          Object.assign(fileReadStream, {
+            _read(this: Readable) {
+              this.push(fileContents);
+              this.push(null);
+            },
+          });
+
+          const nestedPath = path.join(tmpDirPath, 'a', 'b', 'c', 'file.txt');
+
+          file.download({destination: nestedPath}, (err: Error) => {
+            assert.ifError(err);
+            assert.strictEqual(fs.existsSync(nestedPath), true);
+            fs.readFile(nestedPath, (err, tmpFileContents) => {
+              assert.ifError(err);
+              assert.strictEqual(fileContents, tmpFileContents.toString());
+              done();
+            });
+          });
+        });
+      });
+
+      it('should skip write if asked to write a directory object', done => {
+        // Temporary comment: Option 1: Spy on fs methods
+        const mkdirSync = sandbox.spy(fakeFs, 'mkdirSync');
+        const createWriteStream = sandbox.spy(fakeFs, 'createWriteStream');
+
+        tmp.setGracefulCleanup();
+        tmp.dir(async (err, tmpDirPath) => {
+          assert.ifError(err);
+
+          const fileContents = '';
+
+          Object.assign(fileReadStream, {
+            _read(this: Readable) {
+              this.push(fileContents);
+              this.push(null);
+            },
+          });
+
+          const nestedDir = path.join(tmpDirPath, 'a', 'b', 'c', '/');
+
+          file.download(
+            {destination: nestedDir},
+            (err: Error, buffer: Buffer) => {
+              try {
+                assert.ifError(err);
+
+                // Temporary comment: Option 1: Spy on fs methods
+                assert.strictEqual(createWriteStream.callCount, 0);
+                assert.strictEqual(mkdirSync.callCount, 0);
+
+                // Temporary comment: Option 2: Verify that no folder and/or file were created
+                assert.strictEqual(
+                  fs.existsSync(path.dirname(nestedDir)),
+                  false
+                );
+                assert.strictEqual(fs.existsSync(nestedDir), false);
+
+                // Temporary comment: Not very useful since no user would care about an empty buffer. Could remove.
+                assert.strictEqual(buffer.equals(Buffer.alloc(0)), true);
+                done();
+              } catch (e) {
+                done(e);
```

Comment:
Any `assert(false)` will throw an Error, meaning `done()` never gets called and the test times out. Adding the `try catch` guarantees `done()` gets called. Similar to what is described in this Stack overflow discussion: https://stackoverflow.com/questions/66461468/mocha-test-false-assert-timeouts

I'm a bit surprised why this doesn't cause issues in the `fs.readFile` callback assertions in the other tests.

Reply:
This test is remodified into `TransferManager` and this part is no longer needed

Reply:
This test is re-written into `transfer-manager`, so this comment is no longer needed